src/client/public/scripts/utils/track-event.ts

```diff
@@ -0,0 +1,29 @@
+import {EventAnalytics} from '../event-analytics.js';
+import {ClientIDModel} from '../models/client-id.js';
+import {getTrackingId} from '../models/tracking-id.js';
+
+let clientIdModel: ClientIDModel = new ClientIDModel();
+let analytics: EventAnalytics|null = null;
+async function getAnalytics() {
+  if (!analytics) {
+    clientIdModel = new ClientIDModel();
+    analytics = new EventAnalytics(getTrackingId());
+  }
+  return analytics;
+}
+
+export async function trackEvent(eventAction: string, dataSource: string) {
+  try {
+    const clientId = await clientIdModel.getId();
+    const analytics = await getAnalytics();
+    return analytics.trackEvent({
+      clientId,
+      eventAction,
+      eventCategory: dataSource,
```

Comment:
This seems a little odd to me. We will have:
 - window -> automerge-selection-merge
 - window -> automerge-selection-squash
 - serviceworker -> notification-shown

This will make it harder to answer questions such as how many people are using the auto merge feature. Analytics events supports Category -> Action -> Label, which I believe you can filter by any of them.
So we could do:
 - automerge -> available
 - automerge -> selection -> merge
 - automerge -> selection -> squash
 - automerge -> completion
 - notifications -> shown -> review-requested
 - notifications -> shown -> pr-approved

WDYT?

Reply:
SGTM